ai_genomics/analysis/openalex_definition.py

```diff
@@ -0,0 +1,630 @@
+
+    included, excluded = subset_on_concepts(
+        meta_df, concepts_df, selected_concepts, inclusive, return_excluded=True
+    )
+
+    tp = included["arx_ai_conf"].sum()
+    fp = included["arx_no_ai"].sum()
+    tn = excluded["arx_no_ai"].sum()
+    fn = excluded["arx_ai_conf"].sum()
+
+    tp_rate = tp / (tp + tn)
+    fp_rate = fp / (tn + fp)
+    fn_rate = fn / (tn + tp)
+    tn_rate = tn / (tn + fp)
+    precision = tp / (tp + fp)
+    recall = tp / (tp + fn)
+    f1_score = 2 * (precision * recall) / (precision + recall)
+
+    if print_examples:
+        random_sample = [
+            (year, included.query(f"publication_year=={year}").sample(sample))
+            for year in included["publication_year"].unique()
+        ]
+
+        random_excluded = [
+            (year, excluded.query(f"publication_year=={year}").sample(sample))
+            for year in excluded["publication_year"].unique()
+        ]
+
+        for r in random_sample:
+
+            logging.info(r[0])
+            logging.info("===")
+            logging.info("\n")
+            for ind, row in r[1].iterrows():
+
+                logging.info(row["display_name"])
+                logging.info("".join(["-"] * len(row["display_name"])))
+                logging.info(row["work_id"])
+                logging.info(abstracts[row["work_id"]])
+                logging.info("\n")
+
+        for r in random_excluded:
+
+            logging.info("EXCLUDED")
+            logging.info(r[0])
+            logging.info("===")
+            logging.info("\n")
+            for ind, row in r[1].iterrows():
+
+                logging.info(row["display_name"])
+                logging.info("".join(["-"] * len(row["display_name"])))
+                logging.info(row["work_id"])
+                logging.info(abstracts[row["work_id"]])
+                logging.info("\n")
+
+        logging.info("\n \n \n")
+
+    eval_results = {
+        "concepts": ", ".join(
+            ["_".join([k, str(np.round(v, 3))]) for k, v in selected_concepts.items()]
+        ),
+        "num_works": len(included),
+        "tp_rate": tp_rate,
+        "fp_rate": fp_rate,
+        "fn_rate": fn_rate,
+        "tn_rate": tn_rate,
+        "precision": precision,
+        "recall": recall,
+        "f1_score": f1_score,
+    }
+
+    if return_included is True:
+        return eval_results, included
+
+    else:
+        return eval_results
+
+
+def subset_on_concepts(
+    works_meta: pd.DataFrame,
+    works_concepts: pd.DataFrame,
+    selected_concepts: Dict,
+    inclusive: bool = True,
+    return_excluded: bool = True,
+) -> pd.DataFrame:
+    """Subsets the data on the selected concepts
+
+    Args:
+        meta_df: dataframe of works metadata
+        concepts_df: dataframe of openalex concepts
+        abstracts: dictionary of abstracts
+        selected_concepts: dictionary with concepts and score thresholds
+        inclusive: flag to addopt an or or and criterion when selecting papers
```

Comment:
```suggestion
        inclusive: flag to adopt an or or and criterion when selecting papers
```
Looks like there's a word missing from this sentence